### DISCUSSION TITLE
Sharing configurations between multiple states.

### DISCUSSION THREAD

**Discussion by dustinknopoff** [statelyai/xstate#1394] [General] [answered]
Let's say I have 2 states that should handle all the same events, in the same way except for a single event/action.

What's the best way to create the machine config for this?

Right now I have it  as:

```ts
const logicEvents = {
  SET_INDEX: {
    target: "idle",
    actions: ["setIndex"],
  //...
  },
};
```

and in my config:

```ts
loaded: {
              on: {
                ...logicEvents,
                ADD_UNDO_REDO: {
                  actions: ["addUndoRedo"],
                },
              },
            },
            undo_redo: {
              on: {
                ...logicEvents,
              },
            },
```

This works but typescript is unhappy with the typing. (I'm not sure what type `logicEvents` should be.) Or Is there a better way?

**Comment by davidkpiano** [comment, marked as answer]
✏️ Please create a [CodeSandbox (XState TypeScript template)](https://codesandbox.io/s/xstate-typescript-template-6dsws) reproduction of this. Thanks!

**Reply by dustinknopoff** [reply]
I found the type declaration for `on` in the machine configuration `on?: TransitionsConfig<TContext, TEvent>;` and labelled my `logicEvents` to match.
